### ISSUE TITLE
Parsing fails by heredoc with line continuation (escape new line)

### ISSUE THREAD

**Issue by yonta** [hadolint/hadolint#699]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

A Dockefile is valid.

### Actual behavior

Parsing fails.
If I remove line continuation, parsing successes.

### Steps to reproduce the behavior

```console
$ hadolint --version
Haskell Dockerfile Linter v2.7.0-2-g0e5bcdc
$ hadolint Dockerfile
Dockerfile:4:3 unexpected 'a' expecting '#', '\', ADD, ARG, CMD, COPY, ENTRYPOINT, ENV, EXPOSE, FROM, HEALTHCHECK, LABEL, MAINTAINER, ONBUILD, RUN, SHELL, STOPSIGNAL, USER, VOLUME, WORKDIR, a pragma, at least one space, or end of input
```

```console
$  docker run --rm hadolint/hadolint hadolint --version
Haskell Dockerfile Linter v2.6.0-0-g7160964
$ docker run --rm -i hadolint/hadolint < Dockerfile
/dev/stdin:4:3 unexpected 'a' expecting '#', '\', ADD, ARG, CMD, COPY, ENTRYPOINT, ENV, EXPOSE, FROM, HEALTHCHECK, LABEL, MAINTAINER, ONBUILD, RUN, SHELL, STOPSIGNAL, USER, VOLUME, WORKDIR, a pragma, at least one space, or end of input
```

Minimum **Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:1.3-labs
FROM ubuntu
RUN <<EOF
  apt-get update -qq
  apt-get install -yq --no-install-recommends \
    curl=7.64.* gnupg2=2.2.* build-essential=12.6
EOF
```

**Additional environment details (OS, stack version, etc.)**

- OS: Ubuntu in WSL
- stack version: Version 2.7.1, Git revision 8afe0c2932716b0441cf4440d6942c59568b6b19 x86_64 hpack-0.34.4

**Comment by m-ildefons** [comment]
Hi @yonta ,
thanks for reporting this bug. This was caused by me using the [charLiteral](https://hackage.haskell.org/package/megaparsec-5.3.0/docs/Text-Megaparsec-Lexer.html#v:charLiteral) parser where I should have been using the [anySingle](https://hackage.haskell.org/package/megaparsec-5.3.0/docs/Text-Megaparsec-Lexer.html#v:charLiteral) parser.
I've created a PR to fix this.
Best regards, Moritz
